tools/packbeam/packbeam.c

```diff
@@ -39,11 +42,75 @@ static void pad_and_align(FILE *f);
 static void *uncompress_literals(const uint8_t *litT, int size, size_t *uncompressedSize);
 static void add_module_header(FILE *f, const char *module_name, uint32_t flags);
 
+static int do_pack(int argc, char **argv);
+static int do_list(int argc, char **argv);
+
+void usage3(FILE *out, const char *program, const char *msg) {
+    if (!IS_NULL_PTR(msg)) {
+        fprintf(out, "%s\n", msg);
+    }
+    fprintf(out, "Usage: %s [-h] [-l] <avm-file> [<options>]\n", program);
+    fprintf(out, "    -h    Print this help menu.\n");
+    fprintf(out, "    -l    List modules in an AVM file.\n");
+    fprintf(out, "          Without the -l flag, <options> must be: <beam-file> [<beam-file>]*, where\n");
+    fprintf(out, "          <beam-file> is a compiled ERTS beam file, and\n");
+    fprintf(out, "          the first beam file in the list contains an exported start/0 function.\n");
+}
+
+void usage(const char *program) {
+    usage3(stdout, program, NULL);
+}
+
+
 int main(int argc, char **argv)
 {
-    FILE *pack = fopen(argv[1], "w");
+    int opt;
+    
+    const char *action = "pack";
+    while((opt = getopt(argc, argv, "lh")) != -1) {
```

Comment:
space between while keyword and expression: ` while ((opt = getopt(argc, argv, "lh")) != -1) {`

Reply:
copy-pasta from the inter webs.  Fixed.